### PR TITLE
Fix 3472 - overlap in headline of contact journal

### PR DESCRIPTION
**Pull request by ndegendogo**
## Description

This PR fixes a layout glitch in the contact diary overview which was reported at #3472.
See more details at that ticket and the videos.

Root cause was a harmful resizing of the navigation bar in viewWillAppear.
This line is now moved to viewDidLoad.

I took the opportunity to cleanup the code around navigationBar.prefersLargeTitles for the view controllers in the Contact Diary tab. All navigation controllers in that tab are now created with prefersLargeTitles=true from the beginning, then this property needs not be changed again at runtime.
The screenshots show that we have no regression with regard to title size.

I have tested with simulator for iOS 13.7, 14.3 and 14.4.
I have not tested for iOS 12.5.

## Link to Jira

Jira: Internal Tracking ID: [EXPOSUREAPP-9482](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9482)

github: #3472

## Screenshots

The videos show the bug and the fixed version.

https://user-images.githubusercontent.com/6118267/132965886-04536564-ac48-4762-803f-eeafc61d4c22.mov

https://user-images.githubusercontent.com/6118267/132965889-6dbd7f02-cb4a-4518-8436-ebc32940aadd.mov

---
Internal Tracking ID: [EXPOSUREAPP-9482](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9482)

### CODE REVIEW

**Comment by ndegendogo** [comment]
The screenshots show that we have no regression with regard to title size.

Simulator Screen Shot - iPhone 8 - 2021-09-12 at 00 40 01

Simulator Screen Shot - iPhone 8 - 2021-09-12 at 00 41 25

Simulator Screen Shot - iPhone 8 - 2021-09-12 at 00 42 05

Simulator Screen Shot - iPhone 8 - 2021-09-12 at 00 42 52

File: src/xcode/ENA/ENA/Source/Scenes/Containers/NavigationControllerWithLargeTitle.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+			super.init(rootViewController: rootViewController)
+		} else {
+			super.init(nibName: nil, bundle: nil)
+			self.viewControllers = [rootViewController]
+		}
+		setup()
+	}
+
+	// MARK: - Private
+	private func setup() {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Containers/NavigationControllerWithLargeTitle.swift:30]:
```suggestion
	// MARK: - Private
	
	private func setup() {
```

**Review by pascal-brause — APPROVED** [review]
Look good for me, just saw a missing new line 😅.
Thanks again for your contribution!

File: src/xcode/ENA/ENA/Source/Scenes/Containers/NavigationControllerWithLargeTitle.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+
+
+class NavigationControllerWithLargeTitle: UINavigationController {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/Containers/NavigationControllerWithLargeTitle.swift:8]: Would be nice to have a comment here, why we need this custom `UINavigationController`

Comment (ndegendogo) [reply src/xcode/ENA/ENA/Source/Scenes/Containers/NavigationControllerWithLargeTitle.swift:9]: @marcussc I have added the comment.

**Comment by dsarkar** [comment]
Internal Tracking ID: [EXPOSUREAPP-9482](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9482)

**Review by marcussc — APPROVED** [review]

**Pull request merged**